controllers/factory/vmagent.go

```diff
@@ -1102,11 +1169,27 @@ func BuildRemoteWrites(cr *victoriametricsv1beta1.VMAgent, ssCache *scrapesSecre
 		oauth2ClientSecretFile.flagSetting += fmt.Sprintf("%s,", oaSecretKeyFile)
 		oauth2ClientID.flagSetting += fmt.Sprintf("%s,", oaclientID)
 		oauth2Scopes.flagSetting += fmt.Sprintf("%s,", oascopes)
+
+		if rws.HasStreamAggr() {
```

Comment:
It should be used a bit different - define empty value for each remoteWrite flag object. If one of values notNull - flag values would correctly joined. Otherwise flag will be skipped.
e.g.
 ```go
var keepInputVal, dedupIntVal, streamConfVal string
if rws.HasStreamAggr() {
   streamAggrConfig.isNotNull = true
   streamConfVal = rws.AsConfigMapKey(i, "stream-aggr-conf"))
  
  dedupIntVal =  rws.StreamAggrConfig.DedupInterval
 if dedupIntVal != ""{
   streamAggrDedupInterval.isNotNull = true
   
  }
  
   keepInputVal = rws.StreamAggrConfig.KeepInput
   if keepInputVal != "" {
      streamAggrKeepInput.isNotNull = true
    }
}
streamAggrKeepInput.flagSetting += fmt.Sprintf("%s,", streamConfVal)
streamAggrDedupInterval.flagSetting += fmt.Sprintf("%s,",dedupIntVal)

```

 Also, it would be good to add test cases for it.

Reply:
fixed and also added the test